src/storage/core.rs

```diff
@@ -415,54 +417,97 @@ where
     /// Returns entries with matching key
     /// # Errors
     /// Fails after any disk IO errors.
-    pub async fn read_all(&self, key: impl AsRef<K>) -> Result<Vec<Entry>> {
+    pub async fn read_all(&self, key: impl AsRef<K>) -> Result<ReadResult<Vec<Entry>>> {
         let key = key.as_ref();
         let mut all_entries = Vec::new();
         let safe = self.inner.safe.read().await;
         let active_blob = safe
             .active_blob
             .as_ref()
             .ok_or_else(Error::active_blob_not_set)?;
-        if let Some(entries) = active_blob.read().await.read_all_entries(key).await? {
+        let read = active_blob
+            .read()
+            .await
+            .read_all_entries_with_deletion_marker(key)
+            .await?;
+        if read.is_found() {
```

Comment:
Rejecting `ReadResult<>` in `read_all_entries_with_deletion_marker` return type will simplify the code, allowing you to remove this `if` statement.